packages/block-library/src/navigation-link/index.php

```diff
@@ -104,6 +104,14 @@ function block_core_navigation_link_render_submenu_icon() {
  * @return string Returns the post content with the legacy widget added.
  */
 function render_block_core_navigation_link( $attributes, $content, $block ) {
+	// Don't render the block's subtree if it is a draft.
+	if ( is_numeric( $attributes['id'] ) ) {
+		$post = get_post( $attributes['id'] );
```

Comment:
I think this causes category links to stop working. In that scenario, the ID will be set to the category "tag ID", e.g. a numeric value, but `get_post` doesn't check the categories, it checks for posts. This probably went unnoticed because when the category ID is very small (e.g. 2), that ID probably matches the ID of a post, making this check return true.

But if the category ID doesn't overlap with any post ID, then this would cause the category link to stop working. cc @fullofcaffeine

Reply:
We think a solution might be also checking that the type of link is `post`, but I'm not super familiar with the data structure. :)

Reply:
Related issue: https://github.com/WordPress/gutenberg/issues/28712.

Reply:
This was solved by #28892